### DISCUSSION TITLE
When to use SSR with PocketBase

### DISCUSSION THREAD

**Discussion by hmajid2301** [sveltejs/kit#9511] [Q&A] [answered]
Hi All,

I'm don't know frontend web development particularly well. So I have some questions around when I should do something on the client vs the server when using SvelteKit.

I am building an application with SvelteKit and Pocketbase.

The things the make sense to me when to make requests on the server are `load` functions a user see's a page we make the request to fetch the data we need. Rather than waiting the page to load and making a request on the client.
I also use the server for form actions, such as when a user logins or registers.

However the way I have currently setup Pocketbase means I can only authenticate with it on the server side.

So at the moment, when a user say clicks a button.
I make a fetch request on the client to a SvelteKit API Server (`+server.ts`), which then makes a to the pocketbase server.
In my specific case, I deploy my site to Netlify and my PocketBase app. So it feels like there is wasted
network request

Browser -> Netlify Functions -> Fly (PocketBase)

It feels like there is a wasted network request in this case. Say all I want to do is delete something from the database. 

**So I guess my question is when should I make requests on the server and when should I do it on the client.**

 Sorry if this question might be a bit too unspecific. I can provide details about my specific use-case.

**Comment by teemingc** [comment, marked as answer]
Make requests on the server if it involves secrets, otherwise you can fetch from the client.

**Comment by hmajid2301** [comment]
In this case, I can fetch the JWT token from a cookie on the client side is that acceptable ? Or better done on the server?

**Reply by teemingc** [reply]
Yeah, client-side should be fine. Optionally, you can return some data from a layout load function during the initial SSR request. Then you can store this data in a store to use for your client-side requests.
